server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4981,6 +4981,8 @@ private boolean localDirectoryContains(Directory localDirectory, String file, lo
             logger.debug("File {} does not exist in local FS, downloading from remote store", file);
         } catch (IOException e) {
             logger.warn("Exception while reading checksum of file: {}, this can happen if file is corrupted", file);
+            // For any other exception on reading checksum, we delete the file to re-download again
+            store.deleteQuiet(file);
```

Comment:
Can you please add unit/integration test mimicking this issue ?

Reply:
if `deleteQuiet` fails silently , our recoveries also fail silently . Should we just fail the recovery here ?

Reply:
How are we disambiguating between an actual corruption vs a partial download? In case of actual corruption we should fail the recovery and the shard.

Reply:
In both the cases, we would want to delete the file and re-download.